<a href="https://colab.research.google.com/github/disiree31-abel/cs82a-portfolio/blob/main/Module_4_Lab_2_Textbook_Problem_3_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd

df = pd.read_csv("allbacks.csv")

df.head()

,rownames,volume,area,weight,cover
0,1,885,382,800,hb
1,2,1016,468,950,hb
2,3,1125,387,1050,hb
3,4,239,371,350,hb
4,5,701,371,750,hb


In [2]:
df["cover"].value_counts()


,count
cover,
pb,8
hb,7


In [3]:
df["cover"].sort_values().tolist()


['hb',
 'hb',
 'hb',
 'hb',
 'hb',
 'hb',
 'hb',
 'pb',
 'pb',
 'pb',
 'pb',
 'pb',
 'pb',
 'pb',
 'pb']

In [4]:
len(df)


15

In [5]:
sorted_cover = df["cover"].sort_values().tolist()

n = len(sorted_cover)

if n % 2 == 1:
    print("Middle cover:", sorted_cover[n // 2])
else:
    print("Two middle covers:", sorted_cover[n // 2 - 1], "and", sorted_cover[n // 2])



Middle cover: pb


Median Cover Type

The median cover type is 15 when the cover categories are treated as ordered categories. This means that half of the observations are at or below the middle category and half are at or above it.

In [6]:
df["weight"].mean()


np.float64(706.6666666666666)

In [7]:
mean_weight = df["weight"].mean()

print("Mean weight of all books:", mean_weight, "grams")


Mean weight of all books: 706.6666666666666 grams


Mean Weight of All Books

The mean weight of all books is 706.66 grams. This represents the average weight of a book in the bookstore inventory.

In [8]:
df.groupby("cover")["weight"].mean()


,weight
cover,
hb,796.428571
pb,628.125000


In [13]:
group_means = df.groupby("cover")["weight"].mean()

print("Hardback mean weight:", group_means["hb"], "grams")
print("Paperback mean weight:", group_means["pb"], "grams")


Hardback mean weight: 796.4285714285714 grams
Paperback mean weight: 628.125 grams


In [14]:
difference = group_means["hb"] - group_means["pb"]

print("Difference in mean weight:", difference, "grams")


Difference in mean weight: 168.30357142857144 grams


Group Mean Comparison

Hardback books have an average weight of 796.43 grams, while paperback books have an average weight of 628.125 grams. The difference of approximately 168.3035 grams suggests that the type of cover is associated with differences in book weight, although cover type alone does not prove that it causes the difference.

In [15]:
print("Number of books:", len(df))
print("Total weight:", df["weight"].sum())


Number of books: 15
Total weight: 10600


By Hand Verification

The mean is calculated as:

Mean = total weight ÷ number of books

Mean = 10,600 ÷ 15

Mean = 706.67 grams

Python also calculated the mean as 706.67 grams, confirming the result.

In [16]:
print("===== FINAL RESULTS =====")

print("Median cover type:", sorted_cover[len(sorted_cover) // 2])

print("Mean weight of all books:", df["weight"].mean(), "grams")

print("Mean hardback weight:", group_means["hb"], "grams")

print("Mean paperback weight:", group_means["pb"], "grams")

print("Difference:", abs(group_means["hb"] - group_means["pb"]), "grams")


===== FINAL RESULTS =====
Median cover type: pb
Mean weight of all books: 706.6666666666666 grams
Mean hardback weight: 796.4285714285714 grams
Mean paperback weight: 628.125 grams
Difference: 168.30357142857144 grams
